tests/common/devices/sonic.py

```diff
@@ -1545,9 +1545,7 @@ def active_ip_interfaces(self, ip_ifs, ns_arg=""):
                (k.startswith("PortChannel") and not
                 self.is_backend_portchannel(k))
             ):
-                if (v["admin"] == "up" and v["oper_state"] == "up" and
-                        self.ping_v4(v["peer_ipv4"], ns_arg=ns_arg)
-                    ):
+                if (v["admin"] == "up" and v["oper_state"] == "up"):
```

Comment:
Not checking for ping results in flaky tests where sometimes the link doesn't work due to unrelated reasons. Not sure I understood the ARP not being present, ping will resolve ARP if not present. But you still don't care about 'ping()' success then there's another API to get the IP interfaces, which can be used.

Reply:
@smaheshm this api is used only qos as i can see.  Which API can we use instead of this ? One option i can think of if we want to keep ping then pass option in api to check for ping or not so test case can take  the decision.

Regarding `ping will resolve ARP if not present` this will need retry logic. one set of ping does not solve this. 

But I am thinking why qos test case need check about ping ? If traffic fails anyways test case will fail.
Same assumption is there in general for most of test cases.

Reply:
Instead of removing it, add a count option if not already there. QoS tests does config reload, sometimes not all the ports come up or take time to come up. This causes QoS tests to fail, but later when you log in to debug the ports seem to be fine. In cases when not enough 'UP' ports are available the test exits with 'ports not up' reason rather than failing on QoS tests.

Reply:
updated ping count.